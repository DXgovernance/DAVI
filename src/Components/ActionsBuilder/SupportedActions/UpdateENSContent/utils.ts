import contenthash from 'content-hash';
import { MAINNET_ID, LOCALHOST_ID, GOERLI_ID } from 'utils';
import { utils } from 'ethers';

const DEFAULT_NAMEHASH =
  '0x0000000000000000000000000000000000000000000000000000000000000000';

export const convertToNameHash = (name: string) => {
  if (!name) {
    return DEFAULT_NAMEHASH;
  }
  return utils.namehash(name);
};

export const convertToContentHash = (ipfsHash: string) => {
  if (!ipfsHash) {
    return '';
  }
  return contenthash.fromIpfs(ipfsHash);
};

export const convertToIpfsHash = (contentHash: string) => {
  if (!contentHash) return null;
  return contenthash.decode(contentHash);
};

export const getIpfsUrl = (ipfsHash: string) => {
  const ipfsRoot = 'ipfs://';
  return `${ipfsRoot}${ipfsHash}`;
};

export const isValidChainId = (chainId: number) => {
  if (!chainId || chainId === LOCALHOST_ID) {
    return MAINNET_ID;
  } else {
    return chainId;
  }
};

export const isAvailableOnENS = (chainId: number) => {
  const validChainId = isValidChainId(chainId);
  const ensNetworks = [MAINNET_ID, GOERLI_ID];
  return ensNetworks.includes(validChainId);
};
