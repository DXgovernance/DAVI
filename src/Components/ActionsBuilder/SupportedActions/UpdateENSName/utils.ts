import namehash from 'eth-ens-namehash';
import contenthash from 'content-hash';
import { useState, useEffect } from 'react';

export const convertToNameHash = (name: string) => namehash.hash(name);

export const convertToContentHash = (ipfsHash: string) =>
  contenthash.fromIpfs(ipfsHash);

export const convertToIPFSHash = (contentHash: string) => {
  if (!contentHash) return null;
  return contenthash.decode(contentHash);
};

export const useDebounce = (value: string, delay: number) => {
  const [debouncedValue, setDebouncedValue] = useState(value);

  useEffect(() => {
    const handler = setTimeout(() => {
      setDebouncedValue(value);
    }, delay);

    return () => {
      clearTimeout(handler);
    };
  }, [value, delay]);

  return debouncedValue;
};
