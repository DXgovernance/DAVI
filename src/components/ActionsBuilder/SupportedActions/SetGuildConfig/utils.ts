import { FIELDS } from './constants';
import { BigNumber } from 'ethers';
import { SetGuildConfigFields, ControlField } from './types';

export const bn = (n: string | number | BigNumber) => BigNumber.from(n);

type UpdatedValuesReturn = {
  [key in ControlField]: {
    newValue: BigNumber;
    currentValue: BigNumber;
  };
};
export const getUpdatedValues = (
  current: SetGuildConfigFields,
  modifyed: SetGuildConfigFields
): UpdatedValuesReturn | {} => {
  if (!current || !modifyed) return {};
  return Object.keys(current).reduce((acc, key) => {
    if (!FIELDS.map(f => f.name).includes(key)) return acc;
    const currentValue = bn(current[key] || 0);
    const newValue = bn(modifyed[key] || 0);

    if (!currentValue.eq(newValue)) {
      return {
        ...acc,
        [key]: { newValue, currentValue },
      };
    }
    return acc;
  }, {});
};

export const pickUpdatedOrDefaultValue = (
  current: BigNumber,
  modifyed: BigNumber
): BigNumber => {
  if (BigNumber.isBigNumber(current)) {
    if (BigNumber.isBigNumber(modifyed) && !bn(modifyed).eq(current)) {
      return bn(modifyed);
    }
    return bn(current);
  } else if (BigNumber.isBigNumber(modifyed)) {
    return bn(modifyed);
  }
  return bn(0);
};
