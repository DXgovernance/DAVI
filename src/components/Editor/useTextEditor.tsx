import { useEditor } from '@tiptap/react';
import StarterKit from '@tiptap/starter-kit';
import Focus from '@tiptap/extension-focus';
import Highlight from '@tiptap/extension-highlight';
import Placeholder from '@tiptap/extension-placeholder';
import { Editor } from './components';
import TurndownService from 'turndown';
import useLocalStorageWithExpiry from 'hooks/Guilds/useLocalStorageWithExpiry';

const turndownService = new TurndownService();

export const useTextEditor = (
  localPath: string,
  ttlMs: number,
  placeholder: string
) => {
  const [html, onHTMLChange] = useLocalStorageWithExpiry<string>(
    `${localPath}/html`,
    null,
    ttlMs
  );
  const [md, onMdChange] = useLocalStorageWithExpiry<string>(
    `${localPath}/md`,
    null,
    ttlMs
  );
  const clear = () => {
    EditorConfig.commands.clearContent();
    onMdChange('');
    onHTMLChange('');
  };

  const EditorConfig = useEditor({
    content: html ? html : {},
    extensions: [
      StarterKit.configure({
        history: { depth: 10 },
      }),
      Focus.configure({
        className: 'has-focus',
        mode: 'all',
      }),
      Placeholder.configure({
        placeholder,
      }),
      Highlight,
    ],
    onUpdate: ({ editor }) => {
      const html = editor.getHTML();
      if (html) {
        onHTMLChange && onHTMLChange(html);
        onMdChange && onMdChange(turndownService.turndown(html));
        // onJSONChange && onJSONChange(JSON.stringify(editor.getJSON()));
      }
    },
  });

  return { Editor, EditorConfig, html, md, clear };
};
