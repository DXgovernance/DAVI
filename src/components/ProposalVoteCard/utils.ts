import { ConfirmVoteProposalProps, VoteOnProposalProps } from './types';

export const voteOnProposal = ({
  votingPowerAtProposalSnapshot,
  votingPowerAtProposalCurrentSnapshot,
}: VoteOnProposalProps) => {
  const hasNoVotingPower =
    votingPowerAtProposalSnapshot &&
    Number(votingPowerAtProposalSnapshot?.toString()) <= 0;

  const hasVotingPowerAtCurrentSnapshot =
    votingPowerAtProposalCurrentSnapshot &&
    Number(votingPowerAtProposalCurrentSnapshot?.toString()) > 0;
  return {
    hasNoVotingPower,
    hasVotingPowerAtCurrentSnapshot,
  };
};

export const confirmVoteProposal = ({
  proposal,
  contract,
  selectedOption,
  userVotingPower,
  createTransaction,
  cb,
}: ConfirmVoteProposalProps) => {
  createTransaction(
    `Vote on proposal ${proposal?.title}`,
    async () => contract.setVote(proposal?.id, selectedOption, userVotingPower),
    true,
    cb
  );
};

export const getOptionLabel = ({ metadata, optionKey, t }) => {
  const metadataLabel = metadata?.voteOptions?.[optionKey];
  return metadataLabel
    ? metadataLabel
    : Number(optionKey) === 0
    ? t('against', { defaultValue: 'Against' })
    : t('option', { optionKey });
};
