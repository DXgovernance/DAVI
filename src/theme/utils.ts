import { css } from 'styled-components';

export const SUPPORTED_THEMES = {
  DARK: 'DARK',
  LIGHT: 'LIGHT',
};

const MEDIA_WIDTHS = {
  upToSmall: 600,
  upToMedium: 960,
  upToLarge: 1280,
};

export const mediaWidthTemplates = Object.keys(MEDIA_WIDTHS).reduce(
  (accumulator, size) => {
    accumulator[size] = (...args: any[]) => css`
      @media (max-width: ${MEDIA_WIDTHS[size]}px) {
        ${
          // @ts-ignore
          css(...args)
        }
      }
    `;
    return accumulator;
  },
  {}
);
