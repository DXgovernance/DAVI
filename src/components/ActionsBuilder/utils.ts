import { DecodedAction } from './types';
import { Permission } from './types';
import Web3 from 'web3';
import { NULL_SIGNATURE } from 'utils';
import { SupportedAction } from './types';

const web3 = new Web3();

export const getPermissionArgs = (
  decodedActions: DecodedAction[]
): Permission[] => {
  const permissions = decodedActions.map(decodedAction => {
    // Native transfer of ETH does not have a function
    if (
      decodedAction?.decodedCall?.callType ===
        SupportedAction.NATIVE_TRANSFER ||
      decodedAction?.decodedCall?.callType === SupportedAction.RAW_TRANSACTION
    ) {
      return {
        from: decodedAction?.decodedCall?.from,
        to: decodedAction?.decodedCall?.to,
        callType: decodedAction?.decodedCall?.callType,
        functionSignature: NULL_SIGNATURE,
      };
    }
    const functionInputs = decodedAction?.decodedCall?.function?.inputs?.map(
      input => input.type
    );
    const functionName = decodedAction?.decodedCall?.function?.name;
    const functionSignature = web3.eth.abi.encodeFunctionSignature(
      `${functionName}(${functionInputs.join(',')})`
    );
    return {
      from: decodedAction?.decodedCall?.from,
      to: decodedAction?.decodedCall?.to,
      callType: decodedAction?.decodedCall?.callType,
      functionSignature,
    };
  });
  return permissions;
};
