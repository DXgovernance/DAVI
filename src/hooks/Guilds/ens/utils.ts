// import { isSupportedChainId } from 'components/ActionsBuilder/SupportedActions/UpdateENSContent/utils';
import { MAINNET_ID } from 'utils';

export const isAvailableOnENS = (chainId: number) => {
  const ensNetworks = [MAINNET_ID];
  return ensNetworks.includes(chainId);
};
