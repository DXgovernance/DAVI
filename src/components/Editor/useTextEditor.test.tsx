import { useTextEditor } from './useTextEditor';
import { renderHook } from '@testing-library/react-hooks';
import { render } from 'utils/tests';

describe('Editor', () => {
  it('Should match snapshot', () => {
    console.error = jest.fn();
    const { result } = renderHook(() =>
      useTextEditor(`42/create-proposal`, 345600000, 'Enter proposal body')
    );
    const { Editor, EditorConfig } = result.current;
    const { container } = render(<Editor EditorConfig={EditorConfig} />, {
      container: document.body,
    });
    expect(container).toMatchSnapshot();
  });
});
