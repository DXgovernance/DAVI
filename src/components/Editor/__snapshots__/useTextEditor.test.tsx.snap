// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Editor Should match snapshot 1`] = `
.c0 {
  background-color: #1B1D1F;
  border: 1px solid #303338;
  border-radius: 10px;
  color: #BDC0C7;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  max-height: 26rem;
}

.c5 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  overflow-x: hidden;
  overflow-y: auto;
  padding: 0 1rem;
  -webkit-overflow-scrolling: touch;
}

.c5 .ProseMirror {
  outline: none;
  min-height: 200px;
}

.c5 .ProseMirror > * + * {
  margin-top: 0.75em;
}

.c5 .ProseMirror p.is-editor-empty:first-child::before {
  color: rgba(255,255,255,0.5);
  content: attr(data-placeholder);
  float: left;
  height: 0;
  pointer-events: none;
}

.c5 .ProseMirror ul,
.c5 .ProseMirror ol {
  padding: 0 1rem;
}

.c5 .ProseMirror h1,
.c5 .ProseMirror h2,
.c5 .ProseMirror h3,
.c5 .ProseMirror h4,
.c5 .ProseMirror h5,
.c5 .ProseMirror h6 {
  line-height: 1.1;
}

.c5 .ProseMirror code {
  background-color: rgba(48,51,56,0.2);
  color: #303338;
}

.c5 .ProseMirror pre {
  background: #295FF4;
  border-radius: 10px;
  color: #1B1D1F;
  padding: 0.75rem 1rem;
}

.c5 .ProseMirror pre code {
  background: none;
  color: inherit;
  font-size: 0.8rem;
  padding: 0;
}

.c5 .ProseMirror mark {
  background-color: #faf594;
}

.c5 .ProseMirror img {
  height: auto;
  max-width: 100%;
}

.c5 .ProseMirror hr {
  margin: 1rem 0;
}

.c5 .ProseMirror blockquote {
  border-left: 2px solid rgba(41,95,244,0.1);
  padding-left: 1rem;
}

.c5 .ProseMirror hr {
  border: none;
  border-top: 2px solid rgba(41,95,244,0.2);
  margin: 2rem 0;
}

.c5 .ProseMirror ul[data-type='taskList'] {
  list-style: none;
  padding: 0;
}

.c5 .ProseMirror ul[data-type='taskList'] li {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c5 .ProseMirror ul[data-type='taskList'] li > label {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  margin-right: 0.5rem;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c5 .ProseMirror ul[data-type='taskList'] li > div {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-bottom: 1px solid #303338;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  padding: 1rem;
}

.c3 {
  background-color: rgba(41,95,244,0.1);
  height: 1.25rem;
  margin-left: 0.5rem;
  margin-right: 0.75rem;
  width: 2px;
}

.c2 {
  background-color: #1B1D1F;
  border: none;
  border-radius: 0.4rem;
  color: white;
  height: 1.75rem;
  margin-right: 0.25rem;
  padding: 0.25rem;
  width: 1.75rem;
  cursor: pointer;
}

.c2:hover: {
  background-color: white;
  color: #0d0d0d;
}

.c4 {
  background-color: white;
  border: none;
  border-radius: 0.4rem;
  color: #0d0d0d;
  height: 1.75rem;
  margin-right: 0.25rem;
  padding: 0.25rem;
  width: 1.75rem;
  cursor: pointer;
}

.c4:hover: {
  background-color: white;
  color: #0d0d0d;
}

<body>
   
  <div>
    <div
      class="c0"
    >
      <div
        class="c1"
      >
        <button
          class="c2"
          title="Heading 1"
        >
          <svg
            fill="currentColor"
            height="20"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <path
                d="M0 0H24V24H0z"
                fill="none"
              />
              <path
                d="M13 20h-2v-7H4v7H2V4h2v7h7V4h2v16zm8-12v12h-2v-9.796l-2 .536V8.67L19.5 8H21z"
              />
            </g>
          </svg>
        </button>
        <button
          class="c2"
          title="Heading 2"
        >
          <svg
            fill="currentColor"
            height="20"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <path
                d="M0 0H24V24H0z"
                fill="none"
              />
              <path
                d="M4 4v7h7V4h2v16h-2v-7H4v7H2V4h2zm14.5 4c2.071 0 3.75 1.679 3.75 3.75 0 .857-.288 1.648-.772 2.28l-.148.18L18.034 18H22v2h-7v-1.556l4.82-5.546c.268-.307.43-.709.43-1.148 0-.966-.784-1.75-1.75-1.75-.918 0-1.671.707-1.744 1.606l-.006.144h-2C14.75 9.679 16.429 8 18.5 8z"
              />
            </g>
          </svg>
        </button>
        <div
          class="c3"
        />
        <button
          class="c2"
          title="Bold"
        >
          <svg
            fill="currentColor"
            height="20"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M8 11h4.5a2.5 2.5 0 1 0 0-5H8v5zm10 4.5a4.5 4.5 0 0 1-4.5 4.5H6V4h6.5a4.5 4.5 0 0 1 3.256 7.606A4.498 4.498 0 0 1 18 15.5zM8 13v5h5.5a2.5 2.5 0 1 0 0-5H8z"
              />
            </g>
          </svg>
        </button>
        <button
          class="c2"
          title="Italic"
        >
          <svg
            fill="currentColor"
            height="20"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M15 20H7v-2h2.927l2.116-12H9V4h8v2h-2.927l-2.116 12H15z"
              />
            </g>
          </svg>
        </button>
        <button
          class="c2"
          title="Strike"
        >
          <svg
            fill="currentColor"
            height="20"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M17.154 14c.23.516.346 1.09.346 1.72 0 1.342-.524 2.392-1.571 3.147C14.88 19.622 13.433 20 11.586 20c-1.64 0-3.263-.381-4.87-1.144V16.6c1.52.877 3.075 1.316 4.666 1.316 2.551 0 3.83-.732 3.839-2.197a2.21 2.21 0 0 0-.648-1.603l-.12-.117H3v-2h18v2h-3.846zm-4.078-3H7.629a4.086 4.086 0 0 1-.481-.522C6.716 9.92 6.5 9.246 6.5 8.452c0-1.236.466-2.287 1.397-3.153C8.83 4.433 10.271 4 12.222 4c1.471 0 2.879.328 4.222.984v2.152c-1.2-.687-2.515-1.03-3.946-1.03-2.48 0-3.719.782-3.719 2.346 0 .42.218.786.654 1.099.436.313.974.562 1.613.75.62.18 1.297.414 2.03.699z"
              />
            </g>
          </svg>
        </button>
        <div
          class="c3"
        />
        <button
          class="c2"
          title="Code"
        >
          <svg
            fill="currentColor"
            height="20"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M23 12l-7.071 7.071-1.414-1.414L20.172 12l-5.657-5.657 1.414-1.414L23 12zM3.828 12l5.657 5.657-1.414 1.414L1 12l7.071-7.071 1.414 1.414L3.828 12z"
              />
            </g>
          </svg>
        </button>
        <button
          class="c2"
          title="Blockquote"
        >
          <svg
            fill="currentColor"
            height="20"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M4.583 17.321C3.553 16.227 3 15 3 13.011c0-3.5 2.457-6.637 6.03-8.188l.893 1.378c-3.335 1.804-3.987 4.145-4.247 5.621.537-.278 1.24-.375 1.929-.311 1.804.167 3.226 1.648 3.226 3.489a3.5 3.5 0 0 1-3.5 3.5c-1.073 0-2.099-.49-2.748-1.179zm10 0C13.553 16.227 13 15 13 13.011c0-3.5 2.457-6.637 6.03-8.188l.893 1.378c-3.335 1.804-3.987 4.145-4.247 5.621.537-.278 1.24-.375 1.929-.311 1.804.167 3.226 1.648 3.226 3.489a3.5 3.5 0 0 1-3.5 3.5c-1.073 0-2.099-.49-2.748-1.179z"
              />
            </g>
          </svg>
        </button>
        <button
          class="c2"
          title="Highlight"
        >
          <svg
            fill="currentColor"
            height="20"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M15.243 4.515l-6.738 6.737-.707 2.121-1.04 1.041 2.828 2.829 1.04-1.041 2.122-.707 6.737-6.738-4.242-4.242zm6.364 3.535a1 1 0 0 1 0 1.414l-7.779 7.779-2.12.707-1.415 1.414a1 1 0 0 1-1.414 0l-4.243-4.243a1 1 0 0 1 0-1.414l1.414-1.414.707-2.121 7.779-7.779a1 1 0 0 1 1.414 0l5.657 5.657zm-6.364-.707l1.414 1.414-4.95 4.95-1.414-1.414 4.95-4.95zM4.283 16.89l2.828 2.829-1.414 1.414-4.243-1.414 2.828-2.829z"
              />
            </g>
          </svg>
        </button>
        <div
          class="c3"
        />
        <button
          class="c4"
          title="Paragraph"
        >
          <svg
            fill="currentColor"
            height="20"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M12 6v15h-2v-5a6 6 0 1 1 0-12h10v2h-3v15h-2V6h-3zm-2 0a4 4 0 1 0 0 8V6z"
              />
            </g>
          </svg>
        </button>
        <button
          class="c2"
          title="Bullet List"
        >
          <svg
            fill="currentColor"
            height="20"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M8 4h13v2H8V4zM4.5 6.5a1.5 1.5 0 1 1 0-3 1.5 1.5 0 0 1 0 3zm0 7a1.5 1.5 0 1 1 0-3 1.5 1.5 0 0 1 0 3zm0 6.9a1.5 1.5 0 1 1 0-3 1.5 1.5 0 0 1 0 3zM8 11h13v2H8v-2zm0 7h13v2H8v-2z"
              />
            </g>
          </svg>
        </button>
        <button
          class="c2"
          title="Ordered List"
        >
          <svg
            fill="currentColor"
            height="20"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M8 4h13v2H8V4zM5 3v3h1v1H3V6h1V4H3V3h2zM3 14v-2.5h2V11H3v-1h3v2.5H4v.5h2v1H3zm2 5.5H3v-1h2V18H3v-1h3v4H3v-1h2v-.5zM8 11h13v2H8v-2zm0 7h13v2H8v-2z"
              />
            </g>
          </svg>
        </button>
        <div
          class="c3"
        />
        <button
          class="c2"
          title="Horizontal Rule"
        >
          <svg
            fill="currentColor"
            height="20"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M2 11h2v2H2v-2zm4 0h12v2H6v-2zm14 0h2v2h-2v-2z"
              />
            </g>
          </svg>
        </button>
        <div
          class="c3"
        />
        <button
          class="c2"
          title="Undo"
        >
          <svg
            fill="currentColor"
            height="20"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M5.828 7l2.536 2.536L6.95 10.95 2 6l4.95-4.95 1.414 1.414L5.828 5H13a8 8 0 1 1 0 16H4v-2h9a6 6 0 1 0 0-12H5.828z"
              />
            </g>
          </svg>
        </button>
        <button
          class="c2"
          title="Redo"
        >
          <svg
            fill="currentColor"
            height="20"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M18.172 7H11a6 6 0 1 0 0 12h9v2h-9a8 8 0 1 1 0-16h7.172l-2.536-2.536L17.05 1.05 22 6l-4.95 4.95-1.414-1.414L18.172 7z"
              />
            </g>
          </svg>
        </button>
      </div>
      <div
        class="c5"
        data-testid="editor-content"
      >
        <div
          class="ProseMirror"
          contenteditable="true"
          tabindex="0"
          translate="no"
        >
          <p
            class="is-empty is-editor-empty"
            data-placeholder="Enter proposal body"
          >
            <br
              class="ProseMirror-trailingBreak"
            />
          </p>
        </div>
      </div>
    </div>
  </div>
   
</body>
`;
